src/components/Room/Room.js

```diff
@@ -66,24 +83,173 @@ function Room() {
     }
   }
 
+
+  /**
+   * @function roomJoinSubmit - submit form to join a room
+   */
+   async function roomJoinSubmit(roomId, roomPassword, userName) {
+    try {
+      if (roomId === '') {
+        throw new Error('Enter a Room ID');
+      } else if (roomPassword === '') {
+        throw new Error('Enter a Room Password');
+      } else if (userName === '') {
+        throw new Error('Enter a personal Name');
+      }
+    } catch (err) {
+      dispatch(setErrorMessage(err.message));
+      return;
+    }
+
+    let data = {
+      roomId: roomId,
+      roomPassword: roomPassword,
+      userName: userName,
+      userId: store.getState().user.userId,
+      isGuest: store.getState().user.userId == null,
+    };
+
+    // reset room data, add userName to state
+    dispatch(clearRoomData());
+    dispatch(clearChatHistory());
+    dispatch(clearVestibuleData());
```

Comment:
unnecessary

Reply:
might need these actually need to investigate